### PR TITLE
Fix some package lint messages

### PR DESCRIPTION
**Pull request by matsl**
## What

Fix some package lint messages.

## Why

Some or the reaming issues looked small so I gave it a try. 

Some are really hard, impossible!?, to fix since we change function in other packages. I supposed that would require some help from the packages we modify to support some hooks or similar for those changes to be possible cleanly!?  

## Note

Each type of change is in a separate commit except for the ChangeLog and copyright year update that tend to be afterthoughts in my case and comes together as the last commit.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,17 @@
+2022-06-14  Mats Lidell  <matsl@gnu.org>
+
+* hload-path.el (hload-path--make-directory-autoloads)
+    (hload-path--internal-make-directory-autoloads): Use defalias for
+    compatibility function.
+    (hyperb:generate-autoloads): Use it.
+
+* hibtypes.el (cl-lib): Require cl-lib to remove package-lint warning.
+
+* hyperbole.el: Remove defalias for compatilility with obsolete functions
```
Comment (rswgnu) [inline ChangeLog:10]: Compatibility misspelling.

File: ChangeLog [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+    since Emacs 25.1 and outline-flag-region available since 2003.
+
+* hui-em-but.el (hproperty:good-colors): package-lint code formating.
```
Comment (rswgnu) [inline ChangeLog:13]: Formatting

**Review by rswgnu — APPROVED** [review]

File: ChangeLog (hunk 1 above)
Comment (matsl) [reply ChangeLog:10]: :scream:

File: ChangeLog (hunk 2 above)
Comment (matsl) [reply ChangeLog]: I must start with having the spell checker active in the commit window. :pray:

**Pull request merged**